server/filestore.go

```diff
@@ -2613,7 +2613,12 @@ func (fs *fileStore) checkSkipFirstBlock(filter string, wc bool) (int, int) {
 			}
 		})
 	} else if psi, ok := fs.psim.Find(stringToBytes(filter)); ok {
-		start, stop = psi.fblk, psi.lblk
+		if psi.fblk < start {
```

Comment:
I think the original was ok

Reply:
You may be right, but just figured having equal handling of `Find` and `Match` paths eliminates one possible way that we could hit surprising behaviour in the future. 

BTW please feel free to un-mark draft if you want to merge or make changes!

Reply:
Let's revert those then will merge.